app/waf/src/controllers/profiles/http_compress.controller.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright 2019 F5 Networks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Filter, repository} from '@loopback/repository';
+import {
+  post,
+  param,
+  get,
+  getFilterSchemaFor,
+  del,
+  requestBody,
+  HttpErrors,
+  RequestContext,
+  RestBindings,
+} from '@loopback/rest';
+import {ProfileHTTPCompressionRepository} from '../../repositories';
+import {Schema, Response, CollectionResponse} from '..';
+import {BaseController} from '../base.controller';
+import {inject} from '@loopback/core';
+import {ProfileHTTPCompression} from '../../models';
+
+const prefix = '/adcaas/v1/profiles';
+
+const createDesc: string =
+  'ProfileHTTPCompress resource that need to be created';
+
+export class ProfileHTTPCompressionController extends BaseController {
+  constructor(
+    @inject(RestBindings.Http.CONTEXT)
+    protected reqCxt: RequestContext,
+    @repository(ProfileHTTPCompressionRepository)
+    private profileHTTPCompressionRepository: ProfileHTTPCompressionRepository,
+  ) {
+    super(reqCxt);
+  }
+
+  @post(prefix + '/http_compress_profiles', {
+    responses: {
+      '200': Schema.response(
+        ProfileHTTPCompression,
+        'Successfully create HTTP_Compress profile resource',
+      ),
+      '400': Schema.badRequest('Invalid profile resource'),
+      '422': Schema.unprocessableEntity('Unprocessable profile resource'),
+    },
+  })
+  async create(
+    @requestBody(Schema.createRequest(ProfileHTTPCompression, createDesc))
+    reqBody: Partial<ProfileHTTPCompression>,
+  ): Promise<Response> {
+    try {
+      reqBody.tenantId = await this.tenantId;
+      const data = await this.profileHTTPCompressionRepository.create(reqBody);
+      return new Response(ProfileHTTPCompression, data);
+    } catch (error) {
+      throw new HttpErrors.BadRequest(error.message);
```

Comment:
The format error thrown from repository can be handled by controller framework, so we usually needn't catch it in controller handler function.

Reply:
Not sure of this. 
If so, why should we have this in endpointpolicy: https://github.com/F5Networks/f5-adcaas-openstack/blob/7834acc84a78e3fd29a31bb38710f95ddb34e852/app/waf/src/controllers/endpointpolicy.controller.ts#L83 and this in pool: https://github.com/F5Networks/f5-adcaas-openstack/blob/7834acc84a78e3fd29a31bb38710f95ddb34e852/app/waf/src/controllers/pool.controller.ts#L78, 

Could we setup the separate PR to solve such code diffuses?

Reply:
ok